modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3107,9 +3108,7 @@ def dropna(self, **kwargs):
             lib.no_default,
             None,
         )
-        # FIXME: this is a naive workaround for this problem: https://github.com/modin-project/modin/issues/5394
-        # if there are too many partitions then all non-full-axis implementations start acting very badly.
-        # The here threshold is pretty random though it works fine on simple scenarios
+        # The map reduce approach works well for frames with few columnar partitions
```

Comment:
```suggestion
        # The map reduce approach works well for frames with few row or column partitions
```

Reply:
It is not correct. I got the follows results:
If Dataframe has few rows (less than 10^7) the old way show a better performance regardless of the number of columns.
If Dataframe has plenty of rows (more than 10^7) and few columns the map reduce approach is better.
If Dataframe has plenty of rows (more than 10^7) and plenty of columns the old  approach is better again.

But I didn't add any new conditions here, because this is beyond the scope of the current task.